lib/src/rules/prefer_is_not_empty.dart

```diff
@@ -90,17 +91,20 @@ class _Visitor extends SimpleAstVisitor<void> {
         getChildren(propertyTarget, 'isNotEmpty').isEmpty) {
       return;
     }
-    // Should be in PrefixExpression.
-    if (isEmptyAccess.parent is! PrefixExpression) {
-      return;
-    }
-    PrefixExpression prefixExpression =
-        isEmptyAccess.parent as PrefixExpression;
-    // Should be !
-    if (prefixExpression.operator.type != TokenType.BANG) {
-      return;
+
+    // Walk up any parentheses above the isEmpty / isNotEmpty.
+    AstNode isEmptyParent = isEmptyAccess.parent;
```

Comment:
Consider using `isEmptyAccess.unParenthesized`.

Reply:
I'm unable to figure this out... unParenthesized returns the things inside the parens, not outside them.

So in the expression, `(((foo)))`, I am holding onto `foo` (isEmptyAccess), and I'm trying to back out of the parens, but unParenthesized does the opposite.

Reply:
Sorry, yes, I missed that. What you're doing is the right (only) way to do this.